### DISCUSSION TITLE
Modifying API rules makes the browser crash

### DISCUSSION THREAD

**Discussion by synapse** [pocketbase/pocketbase#1053] [Q&A]
The rules inputs are very sluggish and it crashes the browser when typing something or even pasting a complete rule. I am using 0.8.0 rc3

Is there any way to stop crashing or to disable syntax highlight/check?

https://user-images.githubusercontent.com/71137/202469637-3c6f233b-6379-4ff2-91bd-eac8d4427070.mov

**Comment by ganigeorgiev** [comment]
That's strange. I'm not able to reproduce it (both on Chrome 106 and Firefox 105.0.2).

Could you test again in an incognito mode to rule out an incompatibility with some browser extension?

Also, could you try opening the inspect tool and check if there are any JS errors logged?

**Reply by synapse** [reply]
I've also tried **Chrome Version 107.0.5304.110 (Official Build) (arm64)** in incognito and **Firefox 107.0 (64-bit)** in incognito.
In both typing is super slow but it seams that only Chromium based browsers crash.


https://user-images.githubusercontent.com/71137/202476282-a6a69cf9-2cee-4f28-87f9-beacaa0e0a56.mov

https://user-images.githubusercontent.com/71137/202476122-bdc4d435-2df4-4330-b55a-f2b3bbddc931.mov

**Reply by ganigeorgiev** [reply]
@synapse I'm not sure how to reproduce it. Will try to update my Chrome and Firefox and will test again

What operating system are you using, Windows? Is the issue reproducible also with the live demo - https://pocketbase.io/demo?

**Reply by synapse** [reply]
I'm using macOS Monterey 12.3. 
Seams super fluid on the online demo (but unfortunately I could not import my JSON schema and try it out with that)
Could a higher number of relational fields/columns influence the speed?

**Reply by ganigeorgiev** [reply]
> Could a higher number of relational fields/columns influence the speed?

That's possible. Could you attach your collections JSON schema here?

**Reply by synapse** [reply]
Sure here it is: 

[pb_schema.json.zip](https://github.com/pocketbase/pocketbase/files/10032872/pb_schema.json.zip)

**Reply by ganigeorgiev** [reply]
@synapse Thank you. Yes, it is the large number of autocomplete fields that causes it to crash. I'll try to debug it and will see how we can improve it (eventually we may limit the autocomplete fields to 100 or something similar).

**Reply by ganigeorgiev** [reply]
@synapse Could you test again with the [v0.8.0-rc4 release](https://github.com/pocketbase/pocketbase/releases/tag/v0.8.0-rc4)?

**Reply by synapse** [reply]
Hey @ganigeorgiev works like a charm 👍 
Cheers
